lib/_stream_readable.js

```diff
@@ -135,6 +137,7 @@ function readableAddChunk(stream, state, chunk, encoding, addToFront) {
       var e = new Error('stream.unshift() after end event');
       stream.emit('error', e);
     } else {
+      stream.bytesRead += state.objectMode ? 0 : chunk.length;
```

Comment:
There is an edge case, when encoding of `this.push(string)` is same as encoding readable, the string was just pass though.